torchtune/utils/_distributed.py

```diff
@@ -212,3 +214,59 @@ def lora_wrap_fsdp(module: nn.Module, recurse: bool, **kwargs):
         return isinstance(module, tuple(modules_to_wrap))
 
     return lora_wrap_fsdp
+
+
+def get_full_finetune_fsdp_wrap_policy(
+    memory_efficient_wrapping: bool, modules_to_wrap: Set[Type]
+) -> FSDPPolicyType:
+    """
+    Retrieves an FSDP wrapping policy based on the specified ``model_type`` and ``modules_to_wrap``.
+    Specifically, for Llama3 model variants, the policy returned will wrap the model's token embedding
+    and output projection in addition to the modules specified in ``modules_to_wrap`` to maximize memory savings.
+    For all other models, only the modules specified will be wrapped.
+    Args:
+        memory_efficient_wrapping (bool): If ``True``, will also wrap embedding and output projection layers
+        with FSDP.
+        modules_to_wrap (Set[Type]): Set of module types to wrap.
+
+    Note:
+        ``memory_efficient_wrapping`` memory improvements have currently only been verified on llama3 workloads,
+            to provide ~15% memory improvement (when used alongside AC memory efficient wrapping). Other workloads
+            have not been verified and may not see the same improvements.
+    Returns:
+        FSDPPolicyType: Wrapping policy that can be passed into ``FullyShardedDataParallel``.
+    """
+    if memory_efficient_wrapping:
+        return _llama3_full_fsdp_wrap_policy(modules_to_wrap=modules_to_wrap)
+    else:
+        return ModuleWrapPolicy(modules_to_wrap)
+
+
+def _llama3_full_fsdp_wrap_policy(modules_to_wrap: Set[Type]) -> FSDPPolicyType:
```

Comment:
Maybe rename from llama3 based on latest changes?

Reply:
We had a bit of this discussion on the chat thread, but at some point Nicolas called this out and I tend to agree, the return type and associated doc string is not very helpful here - I dont know what ```FSDPPolicyType``` means. Can we add more documentation on this? I don't think most of the users will be at the level of sophistication where this type or how it maps to the policy will be immediately clear.

Reply:
Also bump: are we still naming this based on llama3?

Reply:
Renamed